### PR TITLE
Increase timeout k8s-spot-termination-handler

### PR DESCRIPTION
**Pull request by stafot**
Issue: MM-42644

#### Summary
Increase timeout k8s-spot-termination-handler.
This aims to fix some random failures we noticed on re-provisioning and on e2e clusters.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42644

#### Release Note

```release-note
Increase timeout k8s-spot-termination-handler.
This aims to fix some random failures we noticed on re-provisioning and on e2e clusters.
```

### CODE REVIEW

File: internal/provisioner/kops_provisioner_cluster.go [hunk 1]
```diff
@@ -480,41 +481,50 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
...
-						Path:  "/spec/template/spec/containers/0/env/" + strconv.Itoa(i) + "/value",
-						Value: cluster.ID,
-					}}
+			attempts := 5
+			var daemonSetObj *appsv1.DaemonSet
+			for attempt := 0; attempt < attempts; attempt++ {
+				daemonSetObj, err = k8sClient.Clientset.AppsV1().DaemonSets(namespace).Get(ctx, daemonSet, metav1.GetOptions{})
+				if err != nil && attempt < 5 {
+					time.Sleep(5 * time.Second)
+					continue
```
Comment (Szymongib) [inline internal/provisioner/kops_provisioner_cluster.go:490]: I think the issue is actually here https://github.com/mattermost/mattermost-cloud/pull/624/files#diff-c47cc93d0609e79ef543c8fb97a4e2e1c222380084d5340f6cd8441e75b7fad7R269-R271

We initialize context with a timeout of 61 seconds, we shadow this variable in couple places, but here we probably refer to the context with the 61 seconds timeout, therefore if all other things will not complete in a given time the query will fail with `context deadline exceeded` as we observed it.

I would suggest making new context for each Kubernetes API based on the context we want to have for whole provisioning, which probably should be more than 61 seconds. I think that should solve the issue if not we can proceed with retries.

Comment (stafot) [reply internal/provisioner/kops_provisioner_cluster.go:490]: @Szymongib so to clarify your proposal is to create new context for this or to increase the wait time to 120 f.e?

Comment (Szymongib) [reply internal/provisioner/kops_provisioner_cluster.go:490]: Yeah, so we can start by increasing the wait time, we might need to cleanup/standardize how we handle context there.

File: internal/provisioner/kops_provisioner_cluster.go [hunk 2]
```diff
@@ -474,47 +474,52 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
...
-						k8s.PatchStringValue{
-							Op:    "replace",
-							Path:  "/spec/template/spec/containers/0/env/" + strconv.Itoa(i) + "/value",
-							Value: os.Getenv(model.MattermostChannel),
-						})
-				}
-				if envVar.Name == "MATTERMOST_WEBHOOK" && len(os.Getenv(model.MattermostWebhook)) > 0 {
-					payload = append(payload,
-						k8s.PatchStringValue{
+			if daemonSetObj.Spec.Selector != nil {
```
Comment (Szymongib) [inline internal/provisioner/kops_provisioner_cluster.go:492]: Could you help me understand why we are adding this check?

Comment (stafot) [reply internal/provisioner/kops_provisioner_cluster.go:492]: When in a cluster we don't have spot instances at all this selector expected to be nil. So in this case it is faster to bypass this block and continue to the next steps of provisioning.

**Review by Szymongib — APPROVED** [review]

**Pull request merged**